### PR TITLE
Explain why Nickel is not an embedded DSL

### LINKED ISSUE #931: Rationale: Why is Nickel not a DSL written in Haskell
I’m missing from the [rationale](https://github.com/tweag/nickel/blob/master/RATIONALE.md) the answer to the following question:

> Why not use a DSL from within Haskell instead?

Haskell is a well-established language that has all the features Nickel strives towards (prominently: types, contracts, and effects) but also all the developer tooling and a sufficiently large community of software developers _and_ language researchers. Most importantly, one of Haskell’s purposes is to act as a framework for DSLs.

Wouldn’t it be much cheaper to make Nickel a Haskell DSL?

### PR DESCRIPTION
**Pull request by yannham**
Closes #931

Add the answer from #391, trimmed down and reworded a bit, in the RATIONALE document.

### CODE REVIEW

File: RATIONALE.md [hunk 1]
```diff
@@ -186,6 +192,89 @@ extensible, meaning that third-party may define new effects and implement
...
+Kubernetes appealing: even without functions and types, getting completion with
+documentation in-code is already valuable. This is not possible for a generic
+Haskell LSP, with no knowledge of DSLs such as Nickel.
+
+### Haskell is heavy
+
+One of the possible use-case of Nickel would be to embed Nickel, either as
+binary or linked to it as a C library, into another tool to use (think a cloud
+orchestrator like Terraform). Pulling all of the GHC toolchain to evaluate a few
+hundred lines of lines of configuration sounds overkill. Garbage collection and
```
Comment (vkleen) [inline RATIONALE.md:246]:
```suggestion
hundred lines of configuration sounds overkill. Garbage collection and
```

**Review by vkleen — APPROVED** [review]

File: RATIONALE.md [hunk 2]
```diff
@@ -186,6 +192,89 @@ extensible, meaning that third-party may define new effects and implement
...
 
+## Why Nickel is not a DSL embedded in an existing language
+
+Using an existing language to embed configuration is a common approach. In this
+section, we'll try to answer the question why Nickel isn't a DSL embedded in,
+say, Haskell.
+
+Beside typechecking, the Nix language itself could also be the host of a
+configuration DSL. This is in fact exactly what Nixpkgs and NixOS modules are, a
+DSL embedded into the Nix language.
```
Comment (jneem) [inline RATIONALE.md:203]: This paragraph, while it makes perfect sense when responding to a question from Valentin, seems out of place here.

**Review by jneem — APPROVED** [review]

File: RATIONALE.md (hunk 2 above)
Comment (yannham) [reply RATIONALE.md:203]: Agreed, will get rid of it :+1:

Comment (yannham) [inline RATIONALE.md:203]:
```suggestion
```

**Pull request merged**